contracts/staking/NativeTokenStakingManager.sol

```diff
@@ -8,9 +8,15 @@ pragma solidity 0.8.25;
 import {INativeTokenStakingManager} from "./interfaces/INativeTokenStakingManager.sol";
 import {Address} from "@openzeppelin/contracts@5.0.2/utils/Address.sol";
 import {StakingManager} from "./StakingManager.sol";
+import {StakingManagerSettings} from "./interfaces/IStakingManager.sol";
 
 contract NativeTokenStakingManager is StakingManager, INativeTokenStakingManager {
     using Address for address payable;
+
+    constructor(StakingManagerSettings memory settings) {
```

Comment:
we should avoid pasing through constructor since the contract is meant to be upgradeable

Reply:
I'll wait to merge this PR until after https://github.com/ava-labs/teleporter/pull/477, and update all the constructor/initialize calls then.